contrib/test.sh

```diff
@@ -21,20 +27,29 @@ then
 fi
 
 
-# Use toolchain if explicitly specified
-if [ -n "$TOOLCHAIN" ]
-then
-    alias cargo="cargo +$TOOLCHAIN"
-fi
+echo "********* Testing std *************"
+# Test without any features first.  std is required for tests
+cargo test --verbose --no-default-features --features="std"
 
-# Test without any features first
-cargo test --verbose --no-default-features
+echo "********* Testing default *************"
 # Then test with the default features
 cargo test --verbose
 
+if [ "$DO_NO_STD" = true ]
+then
+echo "********* Testing no-std build *************"
+  # Build no_std, to make sure that cfg(test) doesn't hide any issues
+  cargo build --verbose --features="no-std" --no-default-features
+  # Test no_std
+  cargo test --verbose --features="no-std" --no-default-features
+
+  cargo build --verbose --features="no-std,secp-recovery" --no-default-features
```

Comment:
If you can fix nostd+serde, then can we add a line like 
```
cargo build --verbose --features="no-std $FEATURES" --no-default-features
```
to check that you can turn all the features on at once? (And maybe do a loop over all of them individually, though that seems less essential to me).

Reply:
Added a loop - it caught a couple of things